src/app/services/actions/permission-mode-action-creator.ts

```diff
@@ -10,3 +12,17 @@ export function changeMode(newPerm: PERMISSION_MODE_TYPE): IAction {
         response: newPerm,
     };
 }
+
+export function closePopUp(response: boolean): IAction {
+    return {
+        type: CLOSE_POP_UP_SUCCESS,
+        response,
+    };
+}
+
+export function openPopUp(response: boolean): IAction {
+    return {
+        type: PREPARE_NEXT_POP_UP_SUCCESS,
```

Comment:
Not really sure what `PREPARE_NEXT_POP_SUCCESS` means

Reply:
Like what causes this action type to be triggered?

Reply:
I put in the description where `PREPARE_NEXT_POP_UP_SUCCESS` currently shows up to reset the flag and indicates it will show up the next time the person toggles to app mode again but now it's completely removed

Reply:
sorry i cannot read 
![PuppetAwkwardGIF](https://user-images.githubusercontent.com/55033656/125642809-6c83aa42-aa9a-4c74-a051-559f3c268ff9.gif)

Reply:
lgtm!